core/services/bulletprooftxmanager/types.go

```diff
@@ -0,0 +1,264 @@
+package bulletprooftxmanager
+
+import (
+	"encoding/json"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	gethTypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+// Receipt represents an ethereum receipt.
+//
+// Copied from go-ethereum: https://github.com/ethereum/go-ethereum/blob/ce9a289fa48e0d2593c4aaa7e207c8a5dd3eaa8a/core/types/receipt.go#L50
+//
+// We use our own version because Geth's version specifies various
+// gencodec:"required" fields which cause unhelpful errors when unmarshalling
```

Comment:
weird the gencodec tags are interpreted by the stdlib json package?

Reply:
Its not the tags but the generated Marshal/Unmarshal methods